Tests/PennyTests/Tests/GatewayProcessingTests.swift

```diff
@@ -1,33 +1,97 @@
 @testable import Penny
 @testable import DiscordModels
 @testable import Logging
+import Foundation
 import DiscordGateway
+import ServiceLifecycle
 import Models
+import Shared
 import Testing
 
 extension SerializationNamespace {
     @Suite
-    struct GatewayProcessingTests {
+    final class GatewayProcessingTests: Sendable {
         var responseStorage: FakeResponseStorage { .shared }
         let manager = FakeManager()
-        var context: HandlerContext!
+        let fakeMainService: FakeMainService
+        let context: HandlerContext
+        let mainServiceTask: Task<Void, any Error>
 
         init() async throws {
+            /// Simulate prod
+            setenv("DEPLOYMENT_ENVIRONMENT", "prod", 1)
             /// Disable logging
             LoggingSystem.bootstrapInternal(SwiftLogNoOpLogHandler.init)
-            // reset the storage
+            /// First reset the background runner
+            BackgroundProcessor.sharedForTests = BackgroundProcessor()
+            /// Then reset the storage
             FakeResponseStorage.shared = FakeResponseStorage()
             let fakeMainService = try await FakeMainService(manager: self.manager)
+            self.fakeMainService = fakeMainService
             self.context = fakeMainService.context
-            Task {
+            mainServiceTask = Task<Void, any Error> {
                 try await Penny.start(mainService: fakeMainService)
             }
             await fakeMainService.waitForStateManagerShutdownAndDidShutdownSignals()
         }
+
+        deinit {
+            mainServiceTask.cancel()
+        }
```

Comment:
We cancel the task so it's not really unstructured.

Reply:
This would automatically get cancelled on deinit right?

Reply:
Is that a fact? I guess it would make sense. I couldn't find anything in the proposals and haven't manually tried it yet.

I removed that deinit now.

Reply:
I'm pretty sure otherwise it just wouldn't deinit right?

Reply:
nope.
```swift
@main
struct EntryPoint {
    static func main() async throws {
        print("started")
        let task = Task {
            if (try? await Task.sleep(for: .seconds(60))) == nil {
                print("cancelled")
                fflush(stdout)
            }
        }
        print("over: \(task)")
    }
}
```
results in:
```
started
over: Task<(), Never>(_task: (Opaque Value))
Program ended with exit code: 0
```

Reply:
to be clear, adding `task.cancel(); try await task.value` does make it print "cancelled".